src/Common/UTF8Helpers.cpp

```diff
@@ -94,6 +96,42 @@ size_t computeWidth(const UInt8 * data, size_t size, size_t prefix) noexcept
     size_t rollback = 0;
     for (size_t i = 0; i < size; ++i)
     {
+        /// Quickly skip regular ASCII
+
+#if defined(__SSE2__)
+        const auto lower_bound = _mm_set1_epi8(32);
+        const auto upper_bound = _mm_set1_epi8(126);
+
+        while (i + 15 < size)
+        {
+            __m128i bytes = _mm_loadu_si128(reinterpret_cast<const __m128i *>(&data[i]));
+
+            const uint16_t non_regular_width_mask = _mm_movemask_epi8(
+                _mm_or_si128(
```

Comment:
I wonder if it is makes any sense to lose one of the comarisons by first vector-subtracting 32 and then checking for "less than" against 94.

Reply:
Now we have dependency depth 3:
```
load
compare compare
or
```

If we will do subtraction, it will be:
```
load
subtract
compare
```

The same depth of 3 but less number of instructions.
I'm not sure if it will be beneficial... but it's possible.